dts/bindings/interrupt-controller/shared-irq.yaml

```diff
@@ -6,7 +6,15 @@ include: [interrupt-controller.yaml, base.yaml]
 
 properties:
   interrupts:
-      required: true
+    required: true
 
   label:
-      required: true
+    required: true
+
+  init-priority:
```

Comment:
This would be the first driver to define initialization priority in DTS. While it's not necessary a bad idea such change should be coordinated with the rest of the drivers and the general direction we want to go. Otherwise the usage becomes inconsistent.

Reply:
Good point. For most drivers I don't think there is a real benefit of defining the initialization priority at instance level, so I'll undo this change later. If it is really needed we can discuss it in a further pr.

Reply:
I have restored the configuration of the init priority through Kconfig.